velox/vector/ComplexVector.cpp

```diff
@@ -237,6 +236,71 @@ void RowVector::copy(
   }
 }
 
+void RowVector::copyRanges(
+    const BaseVector* source,
+    const folly::Range<const CopyRange*>& ranges) {
+  if (ranges.empty()) {
+    return;
+  }
+
+  auto minTargetIndex = std::numeric_limits<vector_size_t>::max();
+  auto maxTargetIndex = std::numeric_limits<vector_size_t>::min();
+  for (auto& r : ranges) {
+    minTargetIndex = std::min(minTargetIndex, r.targetIndex);
+    maxTargetIndex = std::max(maxTargetIndex, r.targetIndex + r.count);
+  }
+  SelectivityVector rows(maxTargetIndex);
+  rows.setValidRange(0, minTargetIndex, false);
+  rows.updateBounds();
+  for (auto i = 0; i < children_.size(); ++i) {
+    BaseVector::ensureWritable(
```

Comment:
just curious, will there be any perf impact for calling this for the full range of [minTargetIndex, maxTargetIndex] when the ranges are small and sparsely distributed?  Might be ok with primitive flatvectors are children but what happens when there are complex children

Reply:
The current implementation of array and map vector is not really looking into the selectivity so there is no perf impact.  In theory I could just do [0, maxTargetIndex].